src/blackdoor/cqbe/node/server/RPCHandler.java

```diff
@@ -268,8 +268,19 @@ private JSONObject handleLookupRequest() throws AddressException, UnknownHostExc
 		return responseObject;
 	}
 	
-	private JSONObject handleShutdown(){
-		return null;
+	private void handleShutdown() throws RPCException, IOException{
+		Socket sock = io.getSocket();
+		if(!sock.getInetAddress().isLoopbackAddress() || !sock.getLocalAddress().isLoopbackAddress()){
+			throw new RPCException(JSONRPCError.NON_LO_SHUTDOWN);
+		}
+		io.write(ShutdownRpc.CHALLENGE);
+		String handshake = io.read();
+		if(!handshake.equals(ShutdownRpc.HANDSHAKE)){
+			DBP.printerrorln("Shutdown request was recieved but sender was unable to handshake. "
+					+ "It is possible that this was an attempted loopback spoofing attack.");
+			return;
+		}
+		Node.shutdown();
```

Comment:
Maybe stop accepting first then finish the rpc-queue/wait for all rpc's to be done? Not nessessary, maybe an enhancement for the future